Examples/OneSignalDemo/app/src/main/java/com/onesignal/sdktest/ui/CustomAlertDialogBuilder.java

```diff
@@ -0,0 +1,195 @@
+        mNegativeButtonListener = listener;
+        mNegativeButtonText = text;
+        return this;
+    }
+
+    @Override
+    public CustomAlertDialogBuilder setNeutralButton(CharSequence text, DialogInterface.OnClickListener listener) {
+        mNeutralButtonListener = listener;
+        mNeutralButtonText = text;
+        return this;
+    }
+
+    @Override
+    public CustomAlertDialogBuilder setPositiveButton(CharSequence text, DialogInterface.OnClickListener listener) {
+        mPositiveButtonListener = listener;
+        mPositiveButtonText = text;
+        return this;
+    }
+
+    @Override
+    public CustomAlertDialogBuilder setNegativeButton(int textId, DialogInterface.OnClickListener listener) {
+        setNegativeButton(getContext().getString(textId), listener);
+        return this;
+    }
+
+    @Override
+    public CustomAlertDialogBuilder setNeutralButton(int textId, DialogInterface.OnClickListener listener) {
+        setNeutralButton(getContext().getString(textId), listener);
+        return this;
+    }
+
+    @Override
+    public CustomAlertDialogBuilder setPositiveButton(int textId, DialogInterface.OnClickListener listener) {
+        setPositiveButton(getContext().getString(textId), listener);
+        return this;
+    }
+
+    public CustomAlertDialogBuilder setCanceledOnTouchOutside(boolean cancelOnTouchOutside) {
+        mCancelOnTouchOutside = cancelOnTouchOutside;
+        return this;
+    }
+
+    @Override
+    public AlertDialog create() {
+        throw new UnsupportedOperationException("CustomAlertDialogBuilder.create(): use show() instead..");
+    }
+
+    public Button getPositiveButtonElement() {
+        return positiveButton;
+    }
+
+    public Button getNegativeButtonElement() {
+        return negativeButton;
+    }
+
+    public void setIsCancelable(boolean isCancelable) {
+        this.isCancelable = isCancelable;
+    }
+
+    @Override
+    public AlertDialog show() {
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext(), R.style.WhiteThemeAlertDialog);
+        alertDialogBuilder.setView(view);
+        alertDialogBuilder.setOnKeyListener(new DialogInterface.OnKeyListener() {
+            @Override
+            public boolean onKey(DialogInterface dialog, int keyCode, KeyEvent event) {
+                switch (keyCode) {
+                    case KeyEvent.KEYCODE_BACK:
+                        return !isCancelable;
+                }
+                return false;
+            }
+        });
+        final AlertDialog alertDialog = alertDialogBuilder.create();
+
+        DialogInterface.OnClickListener emptyOnClickListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) { }
+        };
+
+        // Enable buttons (needed for Android 1.6) - otherwise later getButton() returns null
```

Comment:
"needed for Android 1.6"?

Reply:
I stole this from a lib a while back and made it my own for easy `AlertDialog` customization and callbacks, I'd prefer to keep the comments and stuff that came along with it just in case